hazelcast/src/main/java/com/hazelcast/concurrent/lock/LockServiceImpl.java

```diff
@@ -258,10 +258,20 @@ public void commitMigration(PartitionMigrationEvent event) {
             clearLockStoresHavingLesserBackupCountThan(event.getPartitionId(), event.getNewReplicaIndex());
         } else {
             int partitionId = event.getPartitionId();
+            // Local locks are local to the partition and replicaIndex where they have been acquired.
+            // That is the reason they are removed on any partition event on the destination.
+            removeLocalLocks(partitionId);
```

Comment:
If a replica is downgraded `0 -> 1` (in general `0 -> X` where `X > 0`) then that means 0th replica is migrated to somewhere else. So, downgrade from 0 to  X (where `X > 0`) only happens on source, not on destination.

This is very similar to `clearLockStoresHavingLesserBackupCountThan(...)`. Even you can call `removeLocalLocks()` in `clearLockStoresHavingLesserBackupCountThan()`.